core/src/main/java/org/apache/iceberg/util/SortOrderUtil.java

```diff
@@ -37,16 +37,14 @@ private SortOrderUtil() {
   }
 
   public static SortOrder buildSortOrder(Table table) {
-    return buildSortOrder(table.spec(), table.sortOrder());
+    return buildSortOrder(table.schema(), table.spec(), table.sortOrder());
```

Comment:
How was this failing before? The unpartitioned spec isn't usually returned by a table once it has been written and loaded from disk, right?

Reply:
We have a few places where it is used before the table is created. That's where we hit it. Should not affect usual cases.